src/main/java/threads/task/Main.java

```diff
@@ -0,0 +1,171 @@
+            messages.get(getCurrentThreadName()).add(valueOf(++counter));
+        } catch (NullPointerException e) {
+            messages.put(getCurrentThreadName(), new ArrayList<>());
+            messages.get(getCurrentThreadName()).add(valueOf(counter));
+        }
+        System.out.println("                                        write " + getCurrentThreadName() + " value: " + counter);
+    }
+
+    private Map<String, List<String>> getMessagesFromOtherThreads() {
+        Supplier<Map<String, List<String>>> messages = () -> Client.messages
+                .entrySet()
+                .stream()
+                .filter(map -> !map.getKey().equals(getCurrentThreadName()))
+                .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+        return messages.get();
+    }
+
+    public void readNewMessages() {
+        System.out.println(getCurrentThreadName() + " unread messages: " + getMessagesFromOtherThreads().toString());
+        printMessagesNewMessagesFromOtherThreads();
+        clearHistory();
+    }
+
+    private void printMessagesNewMessagesFromOtherThreads() {
+        getMessagesFromOtherThreads()
+                .forEach((key, value) -> System.out.println("From: " + key + " message: " +
+                        value
+                                .stream()
+                                .collect(Collector.of(
+                                        StringBuilder::new,
+                                        (sb, message) -> sb.append(message).append(", "),
+                                        StringBuilder::append,
+                                        sb -> {
+                                            var messages = sb.toString();
+                                            if (messages.endsWith(", "))
+                                                return messages.substring(0, messages.length() - 2);
+                                            return messages;
+                                        }
+                                ))
+                ));
+    }
+
+    private void printMessagesOfInterruptedThreads() {
+        System.out.println("From: " + getCurrentThreadName() + " message: " + messages
+                .get(getCurrentThreadName())
+                .stream()
+                .collect(Collector.of(
+                        StringBuilder::new,
+                        (sb, message) -> sb.append(message).append(", "),
+                        StringBuilder::append,
+                        sb -> {
+                            var messages = sb.toString();
+                            if (messages.endsWith(", "))
+                                return messages.substring(0, messages.length() - 2);
+                            return messages;
+                        }
+                )) + " interrupt ______________________________"
+        );
+    }
+
+    public void isThreadActive() {
+        var isOneOfThreadWroteMoreThan3Messages = getMessagesFromOtherThreads()
```

Comment:
what is the idea of checking all other threads and then-current thread If you can do it at once?